### ISSUE TITLE
讨论模型参数和学习率的关系？欢迎大家指点 | 0.6B pretraining on Skypile subset collapses after one epoch

### ISSUE THREAD

**Issue by zzz1YAO** [jingyaogong/minimind#706]
## 中文版

  首先感谢作者开源这个优秀的 repo。这个项目的初衷很适合在有限算力下体验大模型训练与推理流程，这点我非常认可。

  我目前有一些额外算力，所以尝试把模型扩展到了 0.6B。在数据方面，我使用了中文预训练数据集 Skypile 的一个约 10GB 子集，进
  行预训练。

  我的训练命令如下：
```python
  python trainer/train_pretrain.py \
    --epochs 2 \
    --model_config ./configs/model/pretrain_0_6b.json \
    --max_seq_len 2048 \
    --tokenizer_path ./model \
    --learning_rate 5e-4 \
    --save_interval "$save_step" \
    --log_interval 100 \
    --use_wandb \
    --wandb_project minigpt_small \
    --batch_size "$bsz" \
    --accumulation_steps 4 \
    --data_path ../skypile/pretrain_skypile.jsonl
```
  如果没有特别写出来的参数，我使用的都是脚本默认值。

  但在训练了 1 个 epoch 之后，我发现模型几乎没有任何对话能力，输出也出现了明显退化。比如我输入：

  user> 你好

  模型输出变成了类似下面这样：

  assistant> The following generation flags are not valid and may be ignored: ['temperature', 'top_p']...
  us 20 20 20 20 20 20 20 20 20 20 20...

  从结果来看，模型似乎退化到了重复少量 token 的状态，和正常语言模型行为差异很大。

  我想请教几个问题：

  1. 这个 repo 的预训练配方是否主要是面向 26M 级别模型设计的？
  2. 如果把同样的训练思路直接扩展到 0.6B，是否需要显著调整学习率、max_seq_len、epoch、warmup 或数据配比？
  3. 像我这种情况，更可能是训练超参不适配，还是数据分布本身不适合这个规模的模型？

  如果有类似经验，欢迎大家讨论和指点。也欢迎作者指出我这里的配置是否存在明显问题。

  ———

  ## English version

  First of all, thanks to the author for open-sourcing this excellent repo. The project is a very nice way to explore
  LLM training and inference with limited compute, and I really appreciate that.

  I currently have some extra compute, so I tried scaling the model up to 0.6B. For data, I used about a 10GB subset of
  the Chinese pretraining dataset Skypile and trained the model on it.

  My training command was:
```python
 python trainer/train_pretrain.py \
    --epochs 2 \
    --model_config ./configs/model/pretrain_0_6b.json \
    --max_seq_len 2048 \
    --tokenizer_path ./model \
    --learning_rate 5e-4 \
    --save_interval "$save_step" \
    --log_interval 100 \
    --use_wandb \
    --wandb_project minigpt_small \
    --batch_size "$bsz" \
    --accumulation_steps 4 \
    --data_path ../skypile/pretrain_skypile.jsonl
```

  Any parameters not explicitly mentioned above were left at their default values.

  However, after just 1 epoch of training, the model showed almost no conversational ability and clearly degenerated.
  For example, when I prompted:

  user> 你好

  the output looked like this:

  assistant> The following generation flags are not valid and may be ignored: ['temperature', 'top_p']...
  us 20 20 20 20 20 20 20 20 20 20 20...

  From the output, it seems like the model collapsed into repeating a few tokens, which is far from normal language
  model behavior.

  I would like to ask:

  1. Is the pretraining recipe in this repo mainly designed for 26M-scale models?
  2. If the same training setup is scaled directly to 0.6B, would it require major adjustments to learning rate,
     max_seq_len, number of epochs, warmup, or data mix?
  3. In a case like mine, is this more likely to be a hyperparameter mismatch, or a mismatch between the data
     distribution and the model scale?

  If anyone has similar experience, I’d appreciate any thoughts or suggestions. I’d also be grateful if the author could
  point out any obvious issues in my setup.

**Comment by jingyaogong** [comment]
> 这个 repo 的预训练配方是否主要是面向 26M 级别模型设计的？
如果把同样的训练思路直接扩展到 0.6B，是否需要显著调整学习率、max_seq_len、epoch、warmup 或数据配比？
像我这种情况，更可能是训练超参不适配，还是数据分布本身不适合这个规模的模型？
如果有类似经验，欢迎大家讨论和指点。也欢迎作者指出我这里的配置是否存在明显问题。

在没有更多信息的情况下，我「大概」可以排除以上所有问题

https://github.com/jingyaogong/minimind/blob/88e675dc2c20ecb6c082591323632c60df7f5592/eval_llm.py#L74

应该排查的是你自己定义的 `inference` 的prompt cat规则是什么？训练input和推理input是否完全一致？
